sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/encryption/CosmosEncryptionAsyncDatabase.java

```diff
@@ -125,10 +118,7 @@ public CosmosAsyncClientEncryptionKey getClientEncryptionKey(String id) {
             throw new IllegalArgumentException("clientEncryptionKeyId is null or empty");
         }
 
-        EncryptionKeyWrapProvider encryptionKeyWrapProvider =
-            this.cosmosEncryptionAsyncClient.getEncryptionKeyWrapProvider();
-
-        if (!encryptionKeyWrapProvider.getProviderName().equals(newEncryptionKeyWrapMetadata.getType())) {
+        if (!this.cosmosEncryptionAsyncClient.getKeyEncryptionKeyResolverName().equals(newEncryptionKeyWrapMetadata.getType())) {
             throw new IllegalArgumentException("The EncryptionKeyWrapMetadata Type value does not match with the " +
                 "ProviderName of EncryptionKeyStoreProvider configured on the Client. Please refer to https://aka" +
```

Comment:
**ProviderName of EncryptionKeyStoreProvider** configured on the Client - outdated error message

Reply:
Updated the message